gnomad_qc/v4/create_release/create_release_sites_ht_genomes.py

```diff
@@ -490,26 +495,304 @@ def calculate_concatenate_callstats(
+        - 'meta.project_meta.project_subpop': subpopulation label.
+        - 'meta.subsets': dictionary with subset labels as keys and boolean values.
 
-    :param ht: Table with call stats.
-    :param pops_to_remove: List of populations to be removed.
-    :return: Table with the populations removed.
+    :param ht: Table with the sample metadata.
+    :return: Table with the group membership for each sample to be used for computing
+        allele number (AN) per group.
     """
-    pops_to_remove = hl.literal(pops_to_remove)
+    pop_expr = ht.meta.population_inference.pop
+    sex_expr = ht.meta.sex_imputation.sex_karyotype
+    subpop_expr = ht.meta.project_meta.project_subpop
+
+    hts = []
+    for subset in ["All"] + SUBSETS:
+        if subset in ["tgp", "hgdp"]:
+            subset_pop_expr = subpop_expr
+        else:
+            subset_pop_expr = pop_expr
+
+        # Build strata expressions to get group membership for pop, sex, pop-sex.
+        strata_expr = [
+            {"pop": subset_pop_expr},
+            {"sex": sex_expr},
+            {"pop": subset_pop_expr, "sex": sex_expr},
+        ]
+        # For subsets, add strata expressions to get group membership for subset,
+        # subset-pop, subset-sex, and subset-pop-sex.
+        if subset != "All":
+            subset_expr = hl.or_missing(ht.meta.subsets[subset], subset)
+            strata_expr = [{"subset": subset_expr}] + [
+                {"subset": subset_expr, **x} for x in strata_expr
+            ]
+
+        subset_ht = generate_freq_group_membership_array(
+            ht, strata_expr, remove_zero_sample_groups=True
+        )
+        subset_globals = subset_ht.index_globals()
+        if subset != "All":
+            # The group_membership, freq_meta, and freq_meta_sample_count arrays for
+            # subsets is ordered [adj, raw, subset adj, ...] and we want it to be
+            # [subset adj, subset raw, ...] where subset adj and subset raw have the
+            # same sample grouping.
+            subset_adj_idx = 2
+            subset_ht = subset_ht.annotate(
+                group_membership=hl.array(
+                    [subset_ht.group_membership[subset_adj_idx]]
+                ).extend(subset_ht.group_membership[subset_adj_idx:])
+            )
+            freq_meta_sample_count = hl.array(
+                [subset_globals.freq_meta_sample_count[subset_adj_idx]]
+            ).extend(subset_globals.freq_meta_sample_count[subset_adj_idx:])
+
+            # Modify the freq_meta annotation to match the new group_membership and
+            # freq_meta_sample_count arrays.
+            subset_ht = subset_ht.annotate_globals(
+                freq_meta=hl.array(
+                    [
+                        {"subset": subset, "group": "adj"},
+                        {"subset": subset, "group": "raw"},
+                    ]
+                ).extend(subset_globals.freq_meta[subset_adj_idx + 1 :]),
```

Comment:
because the order is `[adj, raw, subset adj, ...]` and we want `[subset adj, subset raw, ...]` I added subset adj and subset raw, so we want to start after subset adj so 3